### PR TITLE
Adding animations configuration to Donut, Line and Stacked Area Charts

### PR DESCRIPTION
**Pull request by Golodhros**
Adding animations configuration to Donut, Line and Stacked Area Charts

## Description
We wanted to add the option of switch on/off the animations on the charts. In this PR, we adress the charts without them (sparkline had it already).

## Motivation and Context
https://github.com/eventbrite/britecharts/issues/243

## How Has This Been Tested?
Ran tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/charts/donut.js [hunk 1]
```diff
@@ -241,23 +242,38 @@ define(function(require){
...
-                    .attr('fill', getSliceFill)
-                    .on('mouseover', tweenGrowthFactory(externalRadius, 0))
-                    .on('mouseout', tweenGrowthFactory(externalRadius - radiusHoverOffset, pieHoverTransitionDuration))
-                    .transition()
-                    .ease(ease)
-                    .duration(pieDrawingTransitionDuration)
-                    .attrTween('d', tweenLoading);
-
+                if (isAnimated) {
+                    slices.enter()
```
Comment (rwholey-eb) [inline src/charts/donut.js]: what if we stored the value of these outside of the if block? DRY stuff out?

File: src/charts/stacked-area.js [hunk 2]
```diff
@@ -536,35 +537,57 @@ define(function(require){
...
-                .data(layers)
-                .transition()
-                .delay( (_, i) => areaAnimationDelays[i])
-                .duration(areaAnimationDuration)
-                .ease(ease)
-                .attr('d', area)
-                .style('opacity', areaOpacity)
-                .style('fill', ({key}) => categoryColorMap[key]);
+            if (isAnimated) {
+                series = svg.select('.chart-group').selectAll('.layer')
```
Comment (rwholey-eb) [inline src/charts/stacked-area.js]: same here

File: src/charts/donut.js (hunk 1 above)
Comment (Golodhros) [inline src/charts/donut.js]: You mean the first block, before the merge?

Let me try, but I think it won't work...

File: src/charts/stacked-area.js (hunk 2 above)
Comment (Golodhros) [inline src/charts/stacked-area.js]: In this case they are slightly different (layersInitial vs layers), so I cannot extract it.

Comment (rwholey-eb) [inline src/charts/stacked-area.js]: got it. thanks!

**Pull request merged**